src/App.jsx

```diff
@@ -1,10 +1,103 @@
 import './App.css';
+import axios from "axios";
+import world_map from "./assets/world_map.png";
+import React, {useState} from "react";
+import continentColorSetter from "./helpers/continentColorSetter.js";
+import countryInformationText from "./helpers/countryInformationText.jsx";
 
 function App() {
 
+    const [countryData, setCountryData] = useState({});
+    const [searchResult, setSearchResult] = useState({});
+    const [countryQuery, setCountryQuery] = useState("");
+    const [queryError, setQueryError] = useState("");
+
+    async function getCountries() {
+        try {
+            const result = await axios.get("https://restcountries.com/v3.1/all?fields=name,flags,population,continents");
+            result.data.sort((a, b) => {
+                return a.population - b.population;
+            })
+            setCountryData(result);
+        } catch (e) {
+            console.error(e);
+        }
+    }
+
+    async function getCountry(e) {
+        e.preventDefault();
+        setQueryError("");
+        setSearchResult("");
+        try {
+            const result = await axios.get(`https://restcountries.com/v3.1/name/${countryQuery}`);
+            setSearchResult(result.data.slice(0, 1));
+            setCountryQuery("");
+        } catch (error) {
+            setQueryError(`Oops, ${countryQuery} doesn't exist. Please try again.`);
+            setCountryQuery("");
+        }
+    }
+
     return (
         <>
-            Maak je applicatie hier!
+            <header>
+                <img src={world_map} alt="world map"/>
+            </header>
+            <main>
+                <h1>World Regions</h1>
+
+                <h2>Search country information</h2>
+
+                <form onSubmit={getCountry} id="search_form">
+                    <input
+                        type="text"
+                        id="search_input"
+                        value={countryQuery}
+                        onChange={(e) => setCountryQuery(e.target.value)}
+                    />
+                    <button id="search_button" type="submit">Search</button>
+                </form>
+
+                {queryError.length > 0 && <article className="search_result_card" id="error_card">{queryError}</article>}
+
+                {Object.keys(searchResult).length > 0 && <article className="search_result_card">
```

Comment:
Als je de lengte van een array wilt weten, dan hoef je niet eerst de "keys" te pakken. Die functie gebruik je eigenlijk ook bij objecten in plaats van arrays (vandaar ook de prefix "Object." 😉 )
```suggestion
                {searchResult.length > 0 && <article className="search_result_card">
```